### ISSUE TITLE
Cannot create a consistent method resolution order (MRO) for bases Callback, PyTorchLightningPruningCallback

### ISSUE THREAD

**Issue by ZhangYH2020** [jdb78/pytorch-forecasting#1468]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.1.0
- Python version: 3.9.7
- Operating System: win 10

I want to import optimize_hyperparameters using: from pytorch_forecasting.models.temporal_fusion_transformer.tuning import optimize_hyperparameters.

But it gets an error: Cannot create a consistent method resolution order (MRO) for bases Callback, PyTorchLightningPruningCallback

**Comment by sunjin7725** [comment]
Same problem in google colab!
Did you solved @ZhangYH2020 ?

**Comment by Logfather** [comment]
- PyTorch-Forecasting version: 1.0.0
- PyTorch-Lightning: 2.1.2
- PyTorch version: 2.1.1
- Python version: 3.10.13
- Operating System: macOS Sonoma 14.14.1.1
- Chip: M2 Max

Today I had the same issue.

**Comment by kkckk1110** [comment]
I have the same error. Did you fix it? @ZhangYH2020

**Comment by Logfather** [comment]
Hi Zhang, sorry no fix yet

**Comment by ZhangYH2020** [comment]
> Same problem in google colab! Did you solved @ZhangYH2020 ?
No, finally I delete this code, and change hyperparameters manually.

**Comment by ZhangYH2020** [comment]
> I have the same error. Did you fix it? @ZhangYH2020

No, finally I delete this code, and change hyperparameters manually.

**Comment by ruuttt** [comment]
For me it worked to downgrade to optuna version 3.4 and pytorch 2.0.1
`pip install torch==2.0.1 pytorch-lightning==2.0.2 pytorch_forecasting==1.0.0 torchaudio==2.0.2 torchdata==0.6.1 torchtext==0.15.2 torchvision==0.15.2 optuna==3.4`

**Comment by AjinkyaBankar** [comment]
It worked for me after downgrading `optuna==3.4.0` and `torch==2.1.0`.

**Comment by Xinyu-Wu-0000** [comment]
`import` below gave me the same error:
```python
from pytorch_forecasting.models.temporal_fusion_transformer.tuning import (
    optimize_hyperparameters,
)
```

packages (Python 3.10.14 (main, Mar 21 2024, 16:24:04) [GCC 11.2.0] on linux):
```
optuna                       3.6.0
optuna-integration           3.6.0
pytorch-forecasting          1.0.0
pytorch-lightning            2.2.1
pytorch_optimizer            2.12.0
torch                        2.2.1
```

**Comment by narencastellon** [comment]
Hello!!. I also have the same problem... it still hasn't been resolved... or what would be a way to solve it!!!

**Comment by ari62** [comment]
I didn't need to downgrade torch, downgrading optuna worked for me:
`pip install optuna==3.4.0 `
my torch is at 2.3.1

**Comment by aleksmaksimovic** [comment]
> I didn't need to downgrade torch, downgrading optuna worked for me: `pip install optuna==3.4.0 ` my torch is at 2.3.1

Thank you, that worked for me!
